src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -213,40 +213,70 @@ public static <T extends SyntaxElement> T parseStatic(String expr, Iterator<? ex
 		}
 	}
 
+	/**
+	 * Attempts parsing into one of the given syntax element classes.
+	 */
 	@Nullable
 	private <T extends SyntaxElement> T parse(Iterator<? extends SyntaxElementInfo<? extends T>> source) {
+		ParsingStack parsingStack = getParser().getParsingStack();
+
 		ParseLogHandler log = SkriptLogger.startParseLogHandler();
 		try {
+			// Iterate over all SyntaxElementInfos
```

Comment:
It seems there's a lot of unnecessary comments. For most of these, the code and comment seem to be equivalent.

Reply:
Mostly trying to make it readable for people not familiar at all with the parser, so anyone feels comfortable modifying this stuff. Does mean there are a lot of comments, but I'd prefer too many over too little. Would like some more opinions on this